statsmodels/tsa/holtwinters.py

```diff
@@ -430,6 +430,227 @@ def _fmt(x):
+                      \end{cases}\\\\
+            e_{b,t} = \begin{cases}
+                          \frac{e_{l,t}}{l_{t-1}}\quad
+                          \text{if trend is multiplicative}\\\\
+                          e_t\cdot l_{t-1} + e_{l,t}
+                          \quad\text{else}
+                      \end{cases}\\\\
+            e_{s,t} = \begin{cases}
+                          0\quad\text{if seasonal is multiplicative}\\\\
+                          e_{t}(l_{t-1}\circ_b\phi b_{t-1})
+                          \quad\text{else}
+                      \end{cases}
+
+        References
+        ----------
+        .. [1] Hyndman, R.J., & Athanasopoulos, G. (2018) *Forecasting:
+           principles and practice*, 2nd edition, OTexts: Melbourne,
+           Australia. OTexts.com/fpp2. Accessed on February 28th 2020.
+        """
+
+        # check inputs
+        if error in ['additive', 'multiplicative']:
+            error = {'additive': 'add', 'multiplicative': 'mul'}[error]
+        if error not in ['add', 'mul']:
+            raise ValueError("Unknown error model")
+
+        # get Holt-Winters settings and parameters
+        trend = self.model.trend
+        damped = self.model.damped
+        seasonal = self.model.seasonal
+
+        alpha = self.params['smoothing_level']
+        beta = self.params['smoothing_slope']
+        gamma = self.params['smoothing_seasonal']
+        phi = self.params['damping_slope']
+        m = self.model.seasonal_periods
+
+        mul_seasonal = seasonal == "mul"
+        mul_trend = trend == "mul"
+
+        n_params = (
+            2 + 2 * self.model.trending + self.model.seasoning * (m + 1) + damped
+            )
+
+        # set initial values
+        # (notation as in https://otexts.com/fpp2/ets.html)
+        y = np.empty((steps, nsim))
+        l = np.empty((steps + 1, nsim))
+        b = np.empty((steps + 1, nsim))
+        s = np.empty((steps + m, nsim))
+
+        # uses python's index wrapping
+        l[-1,:] = self.level[start]
+        b[-1,:] = self.slope[start]
+        if start == -1:
+            s[-m:,:] = np.tile(self.season[-m:], (nsim, 1)).T
+        else:
+            s[-m:,:] = np.tile(self.season[start-m+1:start+1], (nsim, 1)).T
+
+        # define trend and seasonality operations
+        def operation_and_neutral(op):
+            if op == "mul":
+                return np.multiply, 1
+            else:
+                return np.add, 0
+        op_b, neutral_b = operation_and_neutral(trend)
+        op_s, neutral_s = operation_and_neutral(seasonal)
+
+        if trend is None:
+            b[:,:] = neutral_b
+            phi = 1
+        if seasonal is None:
+            s[:,:] = neutral_s
+        if not damped:
+            phi = 1
+
+        if error == 'add':
+            # estimate error variance from residuals
+            sigma = np.sqrt(np.sum(self.resid**2) / (len(self.resid) - n_params))
+            eps = np.random.randn(steps, nsim) * sigma
```

Comment:
Need to let a user (optionally) pass in a RandomState rather than using the global generator.

Reply:
In addition to supporting RandomState, I think we should also allow a few other cases:

1. User passes in the `eps` they want to use directly (helpful for replication / unit testing)
2. User passes in a scipy distribution function they want to use (allows non-Gaussian errors)
3. (Optional, we can leave this as a TODO) User specifies bootstrapping, in which case we re-sample the in-sample errors.

Reply:
I implemented all these with two new keyword arguments:
* `seed` accepts a `RandomState` or an integer
* `random_errors` can be None, an array of values that are used directly, a scipy distribution function, or `"bootstrap"`.